.github/workflows/ci.yml

```diff
@@ -0,0 +1,196 @@
+name: Tests 1.x
+
+on:
+  push:
+    branches:
+    - 1.x
+  pull_request:
+    branches:
+    - 1.x
+
+jobs:
+  smoke:
+    name: Linux smoke test Py${{ matrix.PYTHON_VERSION }}
+    runs-on: ubuntu-latest
+    env:
+      CI: True
+      PYTHON_VERSION: ${{ matrix.PYTHON_VERSION }}
+      RUNNER_OS: 'ubuntu'
+    strategy:
+      fail-fast: false 
+      matrix:
+        PYTHON_VERSION: ['2.7', '3.6']
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v1
+      - name: Install System Packages
+        run: | 
+          sudo apt-get update
+          sudo apt-get install libegl1-mesa
+      - name: Install Conda
+        uses: goanpeca/setup-miniconda@v1
+        with:
+           activate-environment: test
+           auto-update-conda: true
+           auto-activate-base: false
+           python-version: ${{ matrix.PYTHON_VERSION }} 
+      - name: Install package dependencies
+        shell: bash -l {0}
+        run: conda install --file requirements/posix.txt -y -q
+      - name: Install test dependencies
+        shell: bash -l {0}
+        run: |
+          conda install nomkl -y -q
+          conda install --file requirements/tests.txt -y -q
+      - name: Install Package
+        shell: bash -l {0}
+        run: pip install -e .
+      - name: Show environment information
+        shell: bash -l {0}
+        run: |
+          conda info
+          conda list
+      - name: Run tests
+        shell: bash -l {0}
+        run: xvfb-run --auto-servernum pytest spyder_kernels --cov=spyder_kernels -x -vv
+
+  linux:
```

Comment:
Is it possible to have a yaml file per operating system? I think that way this config would be much easier to read.

Reply:
Yes